// payload/dist/folders/utils/getFolderData.js
import { parseDocumentID } from '../../index.js';
import { getFolderBreadcrumbs } from './getFolderBreadcrumbs.js';
import { queryDocumentsAndFoldersFromJoin } from './getFoldersAndDocumentsFromJoin.js';
import { getOrphanedDocs } from './getOrphanedDocs.js';
/**
 * Query for documents, subfolders and breadcrumbs for a given folder
 */ export const getFolderData = async ({ collectionSlug, folderID: _folderID, req, search })=>{
    const { payload, user } = req;
    const parentFolderID = parseDocumentID({
        id: _folderID,
        collectionSlug: payload.config.folders.slug,
        payload
    });
    const breadcrumbsPromise = getFolderBreadcrumbs({
        folderID: parentFolderID,
        req
    });
    if (parentFolderID) {
        // subfolders and documents are queried together
        const documentAndSubfolderPromise = queryDocumentsAndFoldersFromJoin({
            collectionSlug,
            parentFolderID,
            req
        });
        const [breadcrumbs, documentsAndSubfolders] = await Promise.all([
            breadcrumbsPromise,
            documentAndSubfolderPromise
        ]);
        return {
            breadcrumbs,
            documents: documentsAndSubfolders.documents,
            subfolders: documentsAndSubfolders.subfolders
        };
    } else {
        // subfolders and documents are queried separately
        const subfoldersPromise = getOrphanedDocs({
            collectionSlug: payload.config.folders.slug,
            req,
            search
        });
        const documentsPromise = collectionSlug ? getOrphanedDocs({
            collectionSlug,
            req,
            search
        }) : Promise.resolve([]);
        const [breadcrumbs, subfolders, documents] = await Promise.all([
            breadcrumbsPromise,
            subfoldersPromise,
            documentsPromise
        ]);
        return {
            breadcrumbs,
            documents,
            subfolders
        };
    }
};
